Update HGCAL rechit producer and add dEdx

stahlleiton opened the pull request:
#### PR description:

This PR updates the HGCAL rechit producer to be consistent with TICL and testbeam data, and it also add the dEdx correction for calculating the energy.

#### PR validation:


#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

@IzaakWN @pfs

brusale committed:
implement CLUE algorithm in the barrel

Adds CLUE algorithm implementation in the barrel.
Also the respective producer and plugin factory are introduced, along
with the validation chain.

amecca committed:
[TkAl][PixBary][FIX] extractBaryCentre typo in kwargs for list_branches()

brusale committed:
change doHGCAL flag in fillSimHits

Modify doHGCAL flag. Now the check is made on the collection of
SimHits. This is a solution to a problem for which CaloParticles were
not built in HCAL when the doHGCAL flag was set to true.

brusale committed:
add ticl_barrel procModifier and workflows

brusale committed:
rearrange HLT sequences to include ticl_barrel modifier

amecca committed:
[TkAl][PixBary][FIX] PixelBarycentreAnalyzer: do not try to guess the label

Using the title to guess caused a ConfigFileReadError when it was missing

brusale committed:
wf with CP from PU

brusale committed:
fix NGT scouting and minor fixes

brusale committed:
updated runTheMatrix README

mmusich committed:
AlignmentTracksFromVertexSelector consume di-lepton tracks only if using closest vertex to resonance

mmusich committed:
add hltPixelVertices to event content of TkAlHLTTracks{ZMuMu} alca flavours, add collection of pixeltracks associated to hardest vertex in hltPixelVertices

Commit:
Add energy loss to CalibParams SoA

amecca committed:
[TkAl][PixBary][FIX] create jobs only for the alignments that are selected

smorovic committed:
- RawDataBuffer class for Phase-2 data.
  Due to 32-bit range for sourceIDs, linear vector for mapping all possible ranges would be too large, therefore this class uses std:map (supposed to be ROOT compatible).
  Also, it does one big allocation instead of multiple small ones, but total data size need to be known upfront (which will be the case in DAQ input source).
  std::vector<std::byte> is used to wrap this, but we could also just use standard C++ array allocation.
- add interface and unit test
- start from classdef = 2. Setting 1 reports strange error about mismatch with -1
- SLink definitions are moved to a header in DataFormats

amecca committed:
[TkAl][PixBary] Add "extract" job (->text files) from the output of a "single"

It runs extractBarycentre.py:
* input: ROOT files created by a "single" job
* output: text files that can be used to update the TWiki (csv, markdown)
  and a TeX table
* configurable styles (defaults are used for partition, quality etc.)

Updates in extractBarycentre.py:
* arg parsing outside main()
* support for writing directly to a file (instead of stdout)

[SQUASHME] iterate over all the single jobs

pfs commented:
Excellent - thanks a lot!
Maybe it would be good to run a few standard CMSSW workflows for upgrade to check that nothing is broken and make the PR directly to CMSSW?

mmusich committed:
HLT menu development for 15_0_X (branch for 15_1_X)

These updates correspond to the "V1.2" GRun menu for the 2025 pp data-taking period.
This update includes the integration of the following JIRA tickets:

- [CMSHLT-3568](https://its.cern.ch/jira/browse/CMSHLT-3568) [BPH] New higher acceptance Upsilon HLT path
- [CMSHLT-3570](https://its.cern.ch/jira/browse/CMSHLT-3570) [TRK] HLT-Tracking updates for 2025 menu V1.2: mkFit for doublet recovery
- [CMSHLT-3574](https://its.cern.ch/jira/browse/CMSHLT-3574) [TRK,DQM] TRK POG: keep additional collection in HLT output for RelVal samples
- [CMSHLT-3579](https://its.cern.ch/jira/browse/CMSHLT-3579) [BRIL,Scouting] Add ScoutingPF stream to the LumiScan menu
- [CMSHLT-3585](https://its.cern.ch/jira/browse/CMSHLT-3585) [Scouting,DQM] Addition of hltOnlineMetaDataDigis to DQMOnlineScouting stream
- [CMSHLT-3587](https://its.cern.ch/jira/browse/CMSHLT-3587) [TRK] Restore pre-2025 value of hardCurvCut in Pixel CA

smorovic committed:
- migrate fakeBU data generators to the new RawDataBuffer format
- two new conversion modules

brusale committed:
drop cms type specifications

brusale committed:
add wf 29634.758 to ph2_hlt short matrix

jprendi committed:
new scouting dqm sourceclient

jprendi committed:
adding scouting dqm to the build file for testing

silviodonato committed:
include MUO,JME online scouting DQM, including scouting objects in DQM unit test, use scouting rho

mmusich committed:
introduce .772 (NGT scouting with Nano and Validation)

- fix missing ngtScouting modifier

cmsbuild committed:
Merge pull request #48403 from mmasciov/trackListMergedStripOverlap

Strip overlap check for different product ID

mmusich committed:
fix issue with nanoGen

smorovic committed:
code-checks

smorovic committed:
code-format

eyigitba committed:
Changes to implement sector based mode 7 promotion in EMTF for Run 3 2025

cmsbuild committed:
Merge pull request #48386 from flodamas/MuonSkimForIon_151X

Muon skims for the 2025 light-ion runs

smorovic committed:
fix #ifdef definitions

cmsbuild committed:
Merge pull request #48410 from cms-ngt-hlt/fixBL_linFit_15_1_X

Fix Jacobian for linefit in Brokenline.

Dr15Jones committed:
Changes based on review

- removed unused variables
- updated a comment

michael-pitt committed:
modify from master

michael-pitt committed:
restore code-format

mmusich committed:
move from ROOT file to streamers in scouting tests

Dr15Jones committed:
Update expected test output

Ordering of module calls in begin/end/Job/Stream changed.

cmsbuild committed:
Merge pull request #48216 from leobeltra/simplify_and_improve_macros

Merged SoAView into SoALayout and added heterogeneous deepCopy for PortableCollections

cmsbuild committed:
Merge pull request #48423 from eyigitba/EMTF_promoteMode7Run2025_June25

[L1T] Implementing sector based mode 7 promotion in EMTF for Run 3 2025

grzankatest committed:
Increase histogram bin range for RP State per Lumisection

mmusich committed:
resolve the missing DCSRecords printout in JME DQM for HLT monitoring

kpedro88 committed:
treat file as text even when recognized as binary (because of spurious symbols)

kpedro88 committed:
handle dasgoclient-style syntax and treat = signs correct

kpedro88 committed:
always use ibeos-lfn-sort to prepend paths when calling ibeos das client

kpedro88 committed:
use ibeos setting in all matrix commands

Commit:
Modify some parameters for the no cell option for the V19 scenario

Commit:
Code check

jprendi committed:
changing output directory of the scouting monitor collection to go to ScoutingOnline

Commit:
avoid FSC when making LUT XML payload

flodamas committed:
Fix workflows 4.17 and 4.29

Commit:
Changed the CENT012 Threshold for OO run

Dr15Jones committed:
Improvements based on review

- removed unnecessary includes
- removed redundant checks
- fixed comments

Dr15Jones committed:
Avoid possible exception while processing exception case

- Reuse ModuleSignalSentry for this case.
- Keep past behavior where exception in post signal during begin also resulted in module end not being called.

kpedro88 committed:
simplify grep usage

mmusich committed:
JetAnalyzer: allow possiblity to define the online DQM as output folder

guitargeek committed:
Don't use facade functions in MnMigrad to modify parameter state

These functions will go away and the parameter state object should
be modified directly.

kpedro88 committed:
consistent variable name

smorovic committed:
added notice that this is not the finalized/frozen version until end LS3

kpedro88 committed:
do not apply ibeos-lfn-sort to json output

hqucms committed:
Improve the debugging outputs of HGCalUnpacker.

hqucms committed:
Fix fedId handling.

cmsbuild committed:
Merge pull request #48431 from li-chenyan/CMSSWchenyan

[L1T] Changed the CENT012 Threshold for OO run

cmsbuild committed:
Merge pull request #48430 from flodamas/master

Fix workflows 4.17 and 4.29

alexandertuna committed:
Set torch seeds for reproducible training

cmsbuild committed:
Merge pull request #48421 from smorovic/151x-FEDRawDataBuffer

[DAQ] Phase-2 raw data product

smorovic committed:
add back support for old format to RawEventOutputModuleForBU
DaqFakeReader reverted to FEDRawDataCollection so that this is also tested, as it is in use

mmusich committed:
update hltPathOwners.json with new paths for menu GRun 2025 V1.2

mmusich committed:
smart version bump for release of menu GRun 2025-v1.2 and update of makeOnlineGRun script

mmusich committed:
remove frozen menu 2025 v1.1 and introduce new frozen menu 2025 v1.2

Commit:
Add the new ways to introduce PF cuts frm a file derivd from DB

Commit:
Code check

cmsbuild committed:
Merge pull request #48416 from Dr15Jones/moreRefactorWorker

Refactored how begin/end/Job/Stream are implemented

cmsbuild committed:
Merge pull request #48433 from guitargeek/minuit2_fixup

Don't use facade functions in MnMigrad to modify parameter state

cmsbuild committed:
Merge pull request #48435 from smorovic/151x-FEDRawDataBuffer

[DAQ] add back RawEventOutputModule support for FEDRawDataCollection

mmusich committed:
ScoutingCollectionMonitor add muon "Vtx" collections

- fix the displacedVertices input collections

cmsbuild committed:
Merge pull request #48437 from cms-tsg-storm/HLTdevelopment150X_for151X

HLT menu development for `15_0_X` (8/N)  [`15_1_X`]

cmsbuild committed:
Merge pull request #48436 from bsunanda/Run3-alca254

Run3-alca254 Add the new ways to introduce PF cuts from a file derived from DB

Commit:
Add the comma

mmusich committed:
Add stub of LHCInfoPer* payload inspector

mmusich committed:
add tests for LHCInfo PI

mmusich committed:
re-organize ALCARECOTkAlHLTTracks producers

mmusich committed:
better printout

mmusich committed:
improvements and add unit tests

mmusich committed:
apply stage2 L1T customizations to TrackToTrackComparisonHists

alexandertuna committed:
Remove redundant lines

waredjeb committed:
Incude TICL objects in NanoAOD. SimObject an association maps only in hltNanoValCustomize

hjbossi committed:
Add protection to skip addition FSC channels.

cmsbuild committed:
Merge pull request #48444 from hjbossi/fsc_protection

Add protection to skip addition FSC channels.

cmsbuild committed:
Merge pull request #48393 from akritkbehera/Fix/__getattr__attribute

Updated to use __getitem__

cmsbuild committed:
Merge pull request #48385 from gmelachr/master

Update for BPHNano to solve issues with different compilers

slava77devel committed:
fix the loop range

cmsbuild committed:
Merge pull request #48361 from mmasciov/HLTMTVupdate

Update HLT MultiTrackValidator

kpedro88 committed:
not in function

jprendi committed:
added two displaced vertex collections based on hltScoutingMuonPackerVtx and hltScoutingMuonPackerNoVtx

cmsbuild committed:
Merge pull request #47611 from AdrianoDee/flexible_ca_final

A More Flexible And Lightweight CA

cmsbuild committed:
Merge pull request #48440 from mmusich/mm_fix_TrackToTrackComparisonHists_stage2

apply `stage2` L1T customizations to `TrackToTrackComparisonHists`

cmsbuild committed:
Merge pull request #48454 from trackreco/CMSSW_15_0_0_pre2/is48443-nhits

fix the loop range in mkFit MkFinder hit checks

cmsbuild committed:
Merge pull request #48446 from JHiltbrand/hcalLutManagerIgnoreFSC_151X

Avoid FSC Channels when Making HCAL LUT XML Payloads

cmsbuild committed:
Merge pull request #48449 from hqucms/dev/hgcal-unpacker-update

[HGCAL] Update HGCal unpacker to handle flexible fedId assignment

cmsbuild committed:
Merge pull request #48451 from mmusich/mm_LHCInfo_PI

Introduce a stub of `LHCInfoPer*` Payload Inspector

jprendi committed:
minor bug fix

Co-authored-by: Marco Musich <marco.musich@cern.ch>

cmsbuild committed:
Merge pull request #48398 from jprendi/onlineScoutingDQMclinet

New HLT Scouting DQM Online Client

ftorrresd committed:
Increase WF counter buffer size

elenavernazza committed:
Fix filling of ValueMap for DeepJet scores

akritkbehera committed:
[TrackingTools/KalmanUpdators] Add using declaration to prevent function hiding.

ftorrresd committed:
Fix runTheMatrix.py

cmsbuild committed:
Merge pull request #48425 from cms-ngt-hlt/ev_HLTNano_GenJets

Extend NanoAOD for NGT Phase 2 Scouting with Tau and Gen information

cmsbuild committed:
Merge pull request #48442 from SegmentLinking/reproducible_training

LST notebooks for neural network training: make training reproducible

AdrianoDee committed:
HLT Phase2 Alpaka menu in relval_gpu

fabiocos committed:
Primary4DVertexValidation: update profile plots reco vs PU vtx multiplicity, add histogram for real+fake after selection

elenavernazza committed:
Include MC paths in NanoAOD output

fabiocos committed:
Move normalization of line densities for reco to selection ndof > 4

fabiocos committed:
Add static methods to MTDDetId to test whether is BTL or ETL

fwyzard committed:
Update getCudaCoresPerSM for Blackwell GPUs

cmsbuild committed:
Merge pull request #48428 from bsunanda/Phase2-hgx364S

Phase2-hgx364S Modify some parameters for the no cell option for the V19 scenario

mmusich committed:
remove warnings about UnusedProductsForCanDeleteEarly in the phase-2 HLT menu

cmsbuild committed:
Merge pull request #48391 from tvami/CMSFastGenParticleCandidateProducerCleanup

Remove deprecated genParticleCandidates

cmsbuild committed:
Merge pull request #48250 from jsamudio/dev_LHEPtFilter

Add switch for scalar sum to LHEPtFilter

CeliaFernandez committed:
Change fromVertex parameter from False to True for DGL reconstruction

elenavernazza committed:
Add metrics binned in eta and pt for SimpleTrackValidation

AdrianoDee committed:
Fix for n_layers in CAHitNtupletAlpaka::globalBeginRun

pallabidas committed:
fixing out of bound indices

Commit:
Update L1 Tau isolation LUT

cmsbuild committed:
Merge pull request #48396 from ctdax/master

Fixed G4LorentzVector usage, momentum updating, and energy deposit model in FullModelHadronicProcess.cc

cmsbuild committed:
Merge pull request #48461 from akritkbehera/KalmanUpdators/Overloaded-virtual-warning

[TrackingTools/KalmanUpdators] Add using declaration to prevent funct…

cmsbuild committed:
Merge pull request #48470 from mmusich/mm_dev_removeWarning_from_Ph2_HLTmenu

remove warnings about `UnusedProductsForCanDeleteEarly` in the phase-2 HLT menu

cmsbuild committed:
Merge pull request #48462 from elenavernazza/FixDeepJet

[NGT] Fix filling of ValueMap for DeepJet scores for NanoAOD Phase2 HLT

cmsbuild committed:
Merge pull request #48414 from michael-pitt/AddHeavyIonPythia_master

adds Heavy Ion initialization to Pythia

cmsbuild committed:
Merge pull request #48467 from fwyzard/update_getCudaCoresPerSM

Update getCudaCoresPerSM for Blackwell GPUs

ashleyahram committed:
update vertex smearing parameters for 2025 pO collisions

hjbossi committed:
Change of HF threshold of OO era

cmsbuild committed:
Merge pull request #48483 from hjbossi/HFthresholdChange

Change of HF threshold of OO era

cmsbuild committed:
Merge pull request #48453 from cms-ngt-hlt/scoutingMonitor_addMuonVtx

`ScoutingCollectionMonitor` add muon "Vtx" collections

cmsbuild committed:
Merge pull request #48464 from fabiocos/fc-upvalid20250702

MTD Validation: update vertex multiplicity plots

cmsbuild committed:
Merge pull request #48477 from AdrianoDee/fix_globalBeginRun_CA

Fix for `n_layers` Check in `CAHitNtupletAlpaka::globalBeginRun`

cmsbuild committed:
Merge pull request #48466 from fabiocos/fc-ismtd20250702

MTD DetId: add static methods to MTDDetId to test whether it is BTL or ETL

cmsbuild committed:
Merge pull request #48459 from ftorrresd/nano_wfn

Increase Nano relval WF counter buffer size

cmsbuild committed:
Merge pull request #48475 from CeliaFernandez/dev-updateDGL

Change fromVertex parameter from False to True for DGL reconstruction

cmsbuild committed:
Merge pull request #48474 from mmusich/mm_improve_evtContent_ALCARECOTkAlHLTTracks_v2

Improved Event Content for the `TkAlHLT*` ALCARECO

michael-pitt committed:
add DQM plots for FSC detector

civanch committed:
rename G4 -> CMS

civanch committed:
code-format

cmsbuild committed:
Merge pull request #48480 from ashleyahram/master

update a vertex smearing parameters for 2025 pO MC

cmsbuild committed:
Merge pull request #48482 from Pmeiring/master

Update CALO-L2 Tau Isolation LUT for 2025 post-TS1

elenavernazza committed:
Add configurable parameters for eta and pt bins

cmsbuild committed:
Merge pull request #48487 from michael-pitt/Add_FSC_DQM

add DQM plots for new CMS-FSC detector

mmusich committed:
minor improvements and bug-fixes to LHCInfo PayloadInspector

elenavernazza committed:
Organize output folders

civanch committed:
fixed test problem

mmusich committed:
add a unit test for SimpleTrackValidation

Commit:
Make changes to carry out calibration by skipping some of the runs which are used for other tests

Commit:
Code check

mmusich committed:
silence SiPixelRawToCluster verbose ROC and FED error printouts

cmsbuild committed:
Merge pull request #48488 from civanch/clean_up_custom_physics

[SIM] Clean up custom physics sub-library

cmsbuild committed:
Merge pull request #48463 from AdrianoDee/hlt_phase2_gpu_relval

HLT Phase2 Alpaka menu in `relval_gpu`

cmsbuild committed:
Merge pull request #48478 from pallabidas/Phase-2_GCT_PFCluster_update

[Phase-2 L1T] fixing out of bound indices in Phase2L1CaloPFClusterEmulator

Commit:
Correct the cell-less scenario for the V19 version of HGCal geometry

RoccoA97 committed:
Improve syntax clarity and add parameters description

RoccoA97 committed:
Add again L1 scouting tau collections in test config

cmsbuild committed:
Merge pull request #47859 from brusale/clue-barrel-pr-rebase

TICL-barrel: run CLUE in the barrel calorimeters and first workflows

jsamudio committed:
Fix default for fillDescriptions

Commit:
Add PU monitoring related plots for Spike Killer

lwang046 committed:
update laser monitoring tasks for new HCAL laser system

mmusich committed:
fix HLT TICL reconstruction sequences when ticl_barrel is invoked in presence of the alpaka modifier

mmusich committed:
further clean-up of ticl-barrel related configurations

lwang046 committed:
add LaserMon for pin-diode

amecca committed:
[TkAl][PixBary] Add a PixBary "extract" job to the unit tests

brusale committed:
use barrel to RecHitMap only with procModifier

Pass the barrel input to the RecHitMapProducer only when the ticl_barrel
procModifier is used. This fixes failures in ProdLike wfs.

amecca committed:
[TkAl][PixBary] Update documentation

MartinKlevs committed:
Updated MTD reconstruction track quality BDT

Updated the code for MTDTrackQualityMVA in order to support an updated BDT model with more input parameters. Rewritten the BDT part of MVATrainingNtuple.cc to be more similar to Validation/MtdValidation/plugins/MtdTracksValidation.cc and to save extra parameters.

cmsbuild committed:
Merge pull request #48509 from mmusich/mm_fix_forTiclBarrel

fix HLT TICL reconstruction sequences when `ticl_barrel` is invoked in presence of the `alpaka` modifier

Commit:
pileup profiles for Run3pp13p6TeVSpring25 MC compaigns

mmusich committed:
use a PSet to wrap the list of hits to use in RecHitMapProducer

Commit:
As suggested by Andrea

Commit:
Add non-corrected timing plots in ECAL DQM

Commit:
code format

brusale committed:
Revert "use a PSet to wrap the list of hits to use in RecHitMapProducer"

This reverts commit 0c8751e54f9ab34fd7076415f77f7cc18371a2ab.

cmsbuild committed:
Merge pull request #48493 from bsunanda/Run3-alca255

Run3-alca255 Make changes to carry out calibration by skipping some of the runs which are used for other tests

cmsbuild committed:
Merge pull request #48497 from bsunanda/Phase2-hgx364T

Phase2-hgx364T Correct the cell-less scenario for the V19 version of HGCal geometry

cmsbuild committed:
Merge pull request #48494 from mmusich/mm_silence_SiPixelRawToClusterKernel

silence `SiPixelRawToCluster` verbose ROC and FED error printouts

cmsbuild committed:
Merge pull request #48289 from RoccoA97/bmtf_selection_stream

[L1T] L1 Scouting multi-BX Online Selection based on BMTF stubs

Commit:
Make avaialble PFCuts for the year 2025 and equip CalibPlotProperties to display some of the basic differences between two sets of AlCa i/p files

Commit:
Code check

Commit:
Remove the contradiction

brusale committed:
loose cms type specs and add hitMap comment

mmusich committed:
TkAlHLTTracks*: add onlineMetaDataDigis, fix OT clusters keep statements for Phase-2 and limit ALCARECOTkAlHLTPixelZMuMuVertexTracks collections

Commit:
Add a few more utility methoods in RecHitTools which depend on the chosen detector geometry scenario

Commit:
Code check

mmusich committed:
Update ScoutingMuonTriggerAnalyzer trigger selections:

- do not throw if a trigger expression is not in the list,
- use any DST_PFScouting_ZeroBias* for VdM scans

Dr15Jones committed:
Fix indexing problem in ModuleEventAllocMonitor

cmsbuild committed:
Merge pull request #48517 from brusale/fix_ticlBarrel_prodlike_wfs

fix `ProductNotFound` errors in ProdLike workflows

Dr15Jones committed:
Fix SwitchProducer used with named module

Handle case where module passed to SwitchProducer is also
passed to the Process. We clone the module as-late-as-possible.

Commit:
Provide a metod to extract thicknes of depleted layers of different types of silicon wafers

Commit:
Code check

cmsbuild committed:
Merge pull request #48537 from Dr15Jones/fixSwitchProducer2

Fix SwitchProducer used with named module

cmsbuild committed:
Merge pull request #48536 from Dr15Jones/indexModuleEventAllocMonitor

Fix indexing problem in ModuleEventAllocMonitor

cmsbuild committed:
Merge pull request #48533 from mmusich/mm_improve_evtContent_ALCARECOTkAlHLTTracks_v3

`TkAlHLTTracks*`: futher amendments of Event Content

cmsbuild committed:
Merge pull request #48472 from cms-ngt-hlt/SimpleValidationBinned

Add SimpleTrackValidation binned in eta and pt

cmsbuild committed:
Merge pull request #48513 from CMS-LUMI-POG/master

pileup profiles for Run3pp13p6TeVSpring25 MC compaign

cmsbuild committed:
Merge pull request #48524 from bsunanda/Run3-alca256

Run3-alca256 Make available PFCuts for the year 2025 and equip CalibPlotProperties to display some of the basic differences between two sets of AlCa i/p files

cmsbuild committed:
Merge pull request #48529 from mmusich/mm_patch_ScoutingMuonTriggerAnalyzer

Update `ScoutingMuonTriggerAnalyzer` trigger selections

cmsbuild committed:
Merge pull request #48534 from mmusich/mm_LHCInfo_PI

minor improvements and bug-fixes to `LHCInfo` PayloadInspector

cmsbuild committed:
Merge pull request #48528 from bsunanda/Phase2-hgx364U

Phase2-hgx364U Add a few more utility methoods in RecHitTools which depend on the chosen detector geometry scenario

cmsbuild committed:
Merge pull request #48512 from amecca/add-extractPixBary

[TkAl] Add "extract" jobs for the PixelBarycentre validation in the All-in-One framework

cmsbuild committed:
Merge pull request #48514 from MartinKlevs/updated_MTDTrackQualityMVA_BDT

 MTD reconstruction: updated MTD track quality BDT

cmsbuild committed:
Merge pull request #48538 from bsunanda/Phse2-hgx364V

Phse2-hgx364V Provide a method to extract thicknesses of depleted layers of different types of silicon wafers

Dr15Jones committed:
Switch Concurrency test to use catch2

- added verbose output to try to help catch an infrequent failure.

cmsbuild committed:
Merge pull request #48500 from waredjeb/TICLNanoAOD_CMSSW_15_1_X_2025-07-06-2300

TICL Objects in HLT NanoAOD

cmsbuild committed:
Merge pull request #48506 from kyungminparkdrums/151X_SpikeKillerExtraPlots

Add PU related plots for ECAL Spike Killer monitoring

cmsbuild committed:
Merge pull request #48510 from lwang046/HCAL-LaserFix

HCAL DQM: Update laser monitoring tasks for new laser in hcal calibration events

cmsbuild committed:
Merge pull request #48522 from kyungminparkdrums/151X_nonCorrTimingECAL

Add non-corrected timing plots in ECAL DQM

cmsbuild committed:
Merge pull request #48505 from jsamudio/dev_LHEPtFilter

Fix fillDescriptions default for LHEPtFilter

cmsbuild committed:
Merge pull request #48426 from CTPPS/pps_DQMfixIncreaseLS

PPS: Increase X-axis range for "RP State per Lumisection" DQM plot to 3000 LS

Commit:
Make a test code to study properties of a wafer

Commit:
Code check

Commit:
Equip RecHitTools with a few more utilities to find # of wafer types and the thicknesses in the current geometries

Commit:
Code check

hjbossi committed:
Add NeNe to photon flux options for pythia.

Commit:
Change FitPlot and PlotPropertiies to chck differences between EGamma and AlCaRaw

Commit:
Code check

Commit:
Fix merge conflicts

cmsbuild committed:
Merge pull request #48432 from kpedro88/fix_das_client_ibeos

Fix bugs and logic in ibeos das client

cmsbuild committed:
Merge pull request #48555 from bsunanda/Run3-alca257

Run3-alca257 Change FitPlot and PlotPropertiies to chck differences between EGamma and AlCaRaw

cmsbuild committed:
Merge pull request #48544 from Dr15Jones/verboseConcurrencyTest

Switch Concurrency test to use catch2

cmsbuild committed:
Merge pull request #48547 from bsunanda/Phase2-hgx364W

Phase2-hgx364W Make a test code to study properties of a wafer

cmsbuild committed:
Merge pull request #48551 from bsunanda/Phase2-hgx364X

Phase2-hgx364X Equip RecHitTools with a few more utilities to find # of wafer types and the thicknesses in the current geometries

fwyzard committed:
Disable HIP/ROCm device-side error tests

Dr15Jones committed:
Removed call to TH1::SetDefaultSumw2 in TnPEfficiencyClient

- This call was interfering with fits in other modules
- Fixed memory leak
- Avoid unnecessary extra cloning

Dr15Jones committed:
More robust fitting in PFClient

- only attempt fit if have 3 non-zero bins
- improved memory handling
- use a TH1F rather than a MonitorElement as a temporary

Dr15Jones committed:
DQMStore tracing now gives full path to booked ME

Dr15Jones committed:
Added DQMEDHarvester::fillDescription

This allows inheriting classes with fillDescriptions to be able
to add the additional paramters.

Dr15Jones committed:
Can now set dependencies between DQMEDHarvesters

These two modules were found to depend on histograms created by
other modules in the harvester step

civanch committed:
Iteration_useful_for_gpu

civanch committed:
code format

Dr15Jones committed:
Add dependencies between DQMEDHarvesters

These are needed for workflows 145.408 and 136.731.

fwyzard committed:
Improve the Phase 1 pixel raw to cluster

Improve the Phase 1 pixel raw to cluster heterogeneous algorithm:
  - for each group of duplicate pixels, keep one and discard the rest;
  - fix the numer of nearest neighbours after duplicate removal;
  - include the last pixel of each module in the output;
  - remove unnecessary synchronisation.

cmsbuild committed:
Merge pull request #48576 from civanch/preparation_for_gpu

[SIM] Minor code update for future use of GPU

Dr15Jones committed:
Use FinalWaitTask in FWCore/Concurrency unit tests

The FinalWaitTask is more robust than doing a while loop over the tbb_group::wait call.
Also added SAFE_REQUIRE to protect the call to REQUIRE in more places.

Dr15Jones committed:
Fixed English in the documentation

Dr15Jones committed:
Fix race condition in SerialTaskQueue

There was a race where if the queue was empty right when the running task finished and the task sets m_taskChosen and see that the queue is empty and right then a new task is added and that thread sees m_taskChosen was set, then neither threads will start the next task.

mroguljic committed:
fix regional digi morphing layer 2

cmsbuild committed:
Merge pull request #48580 from fwyzard/fix_alpaka_PixelClustering

Improve the Phase-1 pixel raw to cluster

cmsbuild committed:
Merge pull request #48584 from CMSTrackerDPG/regional_digi_morphing_fix

Fix the list of modules in regional digi morphing

cmsbuild committed:
Merge pull request #48575 from Dr15Jones/fixHarvestingSumw2

Removed call to TH1::SetDefaultSumw2 found in harvesting step

cmsbuild committed:
Merge pull request #48577 from Dr15Jones/fixHarvesterDependencies

Add needed dependencies between some DQMEDHarvesters

cmsbuild committed:
Merge pull request #48563 from fwyzard/disable_ROCm_unit_tests

Disable HIP/ROCm device-side error tests

cmsbuild committed:
Merge pull request #48582 from Dr15Jones/improveConcurrencyTest

Fix race condition in SerialTaskQueue

Commit:
Make appropriate correction to include the missing GE11 chamer in te 2025 simulation geometry description

cmsbuild committed:
Merge pull request #48599 from bsunanda/Run3-gex189

Run3-gex189 Make appropriate corrections to include the missing GE11 chamber in the 2025 simulation geometry description

cmsbuild committed:
Merge pull request #48552 from hjbossi/AddNeNeToPhotonFlux

Add NeNe to photon flux options for pythia.

Commit:
Add HGCal testbeam relval

IzaakWN commented:
Superseded by https://github.com/cms-sw/cmssw/pull/48606. Also see https://gitlab.cern.ch/hgcal-dpg/hgcal-comm/-/issues/60.
